File: result/translate/transcoder/java2cpp/c++/LONGEST_INCREASING_SUBSEQUENCE_1.cpp
int f_gold ( int arr [ ] , int n ) {
  int f_gold [ n ] ;
  int i , j , max = 0 ;
  for ( i = 0 ;
  i < n ;
  i ++ ) {
    f_gold [ i ] = 1 ;
  }
  for ( i = 1 ;
  i < n ;
  i ++ ) for ( j = 0 ;
  j < i ;
  j ++ ) if ( arr [ i ] > arr [ j ] && f_gold [ i ] < f_gold [ j ] + 1 ) f_gold [ i ] = f_gold [ j ] + 1 ;
  for ( i = 0 ;
  i < n ;
  i ++ ) if ( max < f_gold [ i ] ) max = f_gold [ i ] ;
  return max ;
}

File: result/translate/chatgpt/python2java/java/LONGEST_INCREASING_SUBSEQUENCE_1.java
public static int f_gold(int[] arr, int n) {
    int[] lis = new int[n];
    Arrays.fill(lis, 1);
    for (int i = 1; i < n; i++) {
        for (int j = 0; j < i; j++) {
            if (arr[i] > arr[j] && lis[i] < lis[j] + 1) {
                lis[i] = lis[j] + 1;
            }
        }
    }
    int maximum = 0;
    for (int i = 0; i < n; i++) {
        maximum = Math.max(maximum, lis[i]);
    }
    return maximum;
}